typings/tests.ts

```diff
@@ -825,7 +825,29 @@ declare const booleanValue: boolean;
 if (interaction.inGuild()) assertType<Snowflake>(interaction.guildId);
 
 client.on('interactionCreate', async interaction => {
+  if (interaction.inCachedGuild()) {
+    assertType<GuildMember>(interaction.member);
+  } else if (interaction.inRawGuild()) {
+    assertType<APIInteractionGuildMember>(interaction.member);
+  } else {
+    assertType<APIGuildMember | GuildMember | null>(interaction.member);
+  }
+
+  if (interaction.isContextMenu()) {
+    if (interaction.inCachedGuild()) {
+      assertType<Guild>(interaction.guild);
```

Comment:
Can you also assert that the Interaction is still a contextMenu one? Or better yet, assert that all of these checks work? (aka if you first check if it's a ContextMenu one, and you check if it's cached or not, you still have a ContextMenu one, same for raw, etc)